examples/md-flexible/src/configuration/YamlParser.cpp

```diff
@@ -1,386 +1,769 @@
+        description = config.boxMin.description;
+
+        auto tmpNode = node[key];
+        config.boxMin.value = {tmpNode[0].as<double>(), tmpNode[1].as<double>(), tmpNode[2].as<double>()};
+      } else if (key == config.boxMax.name) {
+        expected = "YAML-sequence of three floats. Example: [42, 42, 42].";
+        description = config.boxMax.description;
+
+        auto tmpNode = node[key];
+        config.boxMax.value = {tmpNode[0].as<double>(), tmpNode[1].as<double>(), tmpNode[2].as<double>()};
+      } else if (key == config.subdivideDimension.name) {
+        expected = "YAML-sequence of three ints in [0, 1].";
+        description = config.subdivideDimension.description;
+
+        auto tmpNode = node[key];
+        config.subdivideDimension.value = {tmpNode[0].as<bool>(), tmpNode[1].as<bool>(), tmpNode[2].as<bool>()};
+      } else if (key == config.loadBalancingInterval.name) {
+        expected = "Unsigned Integer";
+        description = config.loadBalancingInterval.description;
+
+        int tmp = node[key].as<int>();
+        if (tmp < 0) {
+          throw std::runtime_error("Load balancing interval must be a positive integer.");
+        }
+
+        config.loadBalancingInterval.value = tmp;
+      } else if (key == config.selectorStrategy.name) {
+        expected = "Exactly one selector strategy out of the possible values.";
+        description = config.selectorStrategy.description;
+
+        std::set<autopas::options::SelectorStrategyOption> parsedOptions;
+        if (node[key].IsSequence()) {
+          if (node[key].size() != 1) {
+            throw std::runtime_error("Pass Exactly one selector strategy.");
+          }
+          parsedOptions = autopas::SelectorStrategyOption::parseOptions(
+              autopas::utils::ArrayUtils::to_string(node[key], "", {"", ""}));
+        } else {
+          parsedOptions = autopas::SelectorStrategyOption::parseOptions(node[key].as<std::string>());
+        }
+        config.selectorStrategy.value = *parsedOptions.begin();
+
+      } else if (key == config.boundaryOption.name) {
+        expected = "YAML-sequence of three possible values.";
+        description = config.boundaryOption.description;
+
+        auto tmpNode = node[key];
+        config.boundaryOption.value = {options::BoundaryTypeOption::parseOptionExact(tmpNode[0].as<std::string>()),
+                                       options::BoundaryTypeOption::parseOptionExact(tmpNode[1].as<std::string>()),
+                                       options::BoundaryTypeOption::parseOptionExact(tmpNode[2].as<std::string>())};
+      } else if (key == config.cutoff.name) {
+        expected = "Positive floating point value > 0.";
+        description = config.cutoff.description;
+
+        double tmp = node[key].as<double>();
+        if (tmp <= 0) {
+          throw std::runtime_error("Cutoff has to be > 0!");
+        }
+
+        config.cutoff.value = tmp;
```

Comment:
Sorry I'm really no fan of variables called `tmp` 😝
Why not just:
```suggestion
        config.cutoff.value = node[key].as<double>();
        if (config.cutoff.value <= 0) {
          throw std::runtime_error("Cutoff has to be > 0!");
        }
```

Reply:
Same pattern follows multiple times

Reply:
Removed all tmp-vars.